fahrschein-opentracing/src/main/java/org/zalando/fahrschein/opentracing/OpenTracingWrapper.java

```diff
@@ -0,0 +1,141 @@
+	 * @param spanContext the span context
+	 * @return the nakadi context that can be added to the metadata of the Nakadi
+	 *         event
+	 */
+	public static Map<String, String> convertSpanContext(Tracer tracer, SpanContext spanContext) {
+		Map<String, String> nakadiContext = new HashMap<>();
+		tracer.inject(spanContext, Format.Builtin.TEXT_MAP_INJECT, new TextMapInjectAdapter(nakadiContext));
+		return nakadiContext;
+	}
+
+	/**
+	 * Extracts a SpanContext from the given event.
+	 * 
+	 * @param tracer the tracer instance
+	 * @param event  the event
+	 * @return the SpanContext or <code>null</code> if no context was propagated
+	 */
+	protected static SpanContext extractSpanContext(Tracer tracer, Event event) {
+		Metadata metadata = event.getMetadata();
+		Map<String, String> carrierData = metadata.getSpanCtx();
+
+		return carrierData == null || carrierData.isEmpty() ? null
+				: tracer.extract(Format.Builtin.TEXT_MAP_EXTRACT, new TextMapExtractAdapter(carrierData));
+	}
+
+	/**
+	 * Creates a newly started span instance using the span context in the nakadi
+	 * event as parent.
+	 * 
+	 * @param tracer    the tracer instance
+	 * @param operation the operation of the span
+	 * @param event     the nakadi event
+	 * @return the newly-started Span instance, which has *not* been automatically
+	 *         registered via the {@link ScopeManager}
+	 */
+	protected static Span createSpan(Tracer tracer, String operation, Event event) {
+		SpanContext spanContext = extractSpanContext(tracer, event);
+
+		Span span = null;
+		if (spanContext != null) {
+			span = tracer.buildSpan(operation).asChildOf(spanContext)
+					.withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CONSUMER).start();
+		} else {
+			span = tracer.buildSpan(operation).withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CONSUMER).start();
+		}
+		return span;
+	}
+
+	/**
+	 * A wrapper that handles opentracing around the given function. You need to
+	 * pass in the function that either does the event handling or calls the event
+	 * handling service. The wrapper instance sets up the Span around this
+	 * invocation.
+	 * 
+	 * @param <R>   the return type
+	 * @param <T>   the event type
+	 * @param event the event
+	 * @param f     the function (e.g. lambda)
+	 * @return the result from the function
+	 */
+	public <R, T extends Event> R process(T event, Function<T, R> f) {
+		Span span = createSpan(tracer, operation, event);
+
+		try (Scope scope = tracer.activateSpan(span)) {
+			return f.apply(event);
+		} finally {
+			span.finish();
+		}
+	}
+
+	/**
+	 * A wrapper that handles OpenTracing around the given consumer. You need to
+	 * pass in a consumer that does the event handling or calls the event handling
+	 * service. The wrapper instance sets up the Span around this invocation.
+	 * 
+	 * @param <T>   the event type
+	 * @param event the event
+	 * @param c     the consumer
+	 */
+	public <T extends Event> void process(T event, Consumer<T> c) {
```

Comment:
At the moment, this code is not exercised in tests nor demo code or README, and there's no equivalent in the OpenTelemetryWrapper. Is this a left-over or meant to be used?

Reply:
the code is mainly to wrap the event processing
```
OpenTracingWrapper.process(event, this::doMainWork);
```

Reply:
Could you add the same into the OpenTelemetryWrapper as well?

Reply:
Please also document these methods in the README